athena-jdbc/src/main/java/com/amazonaws/athena/connectors/jdbc/manager/JdbcMetadataHandler.java

```diff
@@ -218,13 +219,46 @@ public GetTableResponse doGetTable(final BlockAllocator blockAllocator, final Ge
     {
         try (Connection connection = jdbcConnectionFactory.getConnection(getCredentialProvider())) {
             Schema partitionSchema = getPartitionSchema(getTableRequest.getCatalogName());
-            return new GetTableResponse(getTableRequest.getCatalogName(), getTableRequest.getTableName(), getSchema(connection, getTableRequest.getTableName(), partitionSchema),
-                    partitionSchema.getFields().stream().map(Field::getName).collect(Collectors.toSet()));
+            Optional<Schema> schema = getSchema(connection, getTableRequest.getTableName(), partitionSchema);
+            if (schema.isPresent()) {
+                return new GetTableResponse(getTableRequest.getCatalogName(), getTableRequest.getTableName(), schema.get(),
+                        partitionSchema.getFields().stream().map(Field::getName).collect(Collectors.toSet()));
+            }
+            else {
+                LOGGER.info("Table {} not found.  Falling back to case insensitive search.", getTableRequest.getTableName());
+
+                TableName caseInsensitiveTableMatch = caseInsensitiveTableSearch(connection, getTableRequest.getTableName().getSchemaName(),
+                        getTableRequest.getTableName().getTableName());
+                Schema caseInsensitiveSchemaMatch = getSchema(connection, caseInsensitiveTableMatch, partitionSchema)
+                        .orElseThrow(() -> new RuntimeException(String.format("Could not find table %s", getTableRequest.getTableName())));
+
+                return new GetTableResponse(getTableRequest.getCatalogName(), caseInsensitiveTableMatch, caseInsensitiveSchemaMatch,
+                        partitionSchema.getFields().stream().map(Field::getName).collect(Collectors.toSet()));
+            }
         }
     }
 
-    private Schema getSchema(Connection jdbcConnection, TableName tableName, Schema partitionSchema)
-            throws SQLException
+    private TableName caseInsensitiveTableSearch(Connection connection, final String databaseName,
+                                                     final String tableName) throws Exception
+    {
+        List<TableName> tables = listTables(connection, databaseName)
```

Comment:
For other readers, we can't benefit from pagination here because we can't put LIMIT on the tables we return from the jdbc query, so this is actually the right way to do this